src/test/java/org/prebid/server/ApplicationTest.java

```diff
@@ -376,63 +764,234 @@ public void ampShouldReturnTargeting() throws IOException, JSONException {
+                .body(jsonFrom("auction/facebook/test-auction-facebook-request.json"))
+                .post("/auction");
 
-        // appnexus bid response for ad unit 4
-        wireMockRule.stubFor(post(urlPathEqualTo("/appnexus-exchange"))
-                .withRequestBody(equalToJson(jsonFrom("auction/test-appnexus-bid-request-1.json")))
-                .willReturn(aResponse().withBody(jsonFrom("auction/test-appnexus-bid-response-1.json"))));
+        // then
+        assertThat(response.header("Cache-Control")).isEqualTo("no-cache, no-store, must-revalidate");
+        assertThat(response.header("Pragma")).isEqualTo("no-cache");
+        assertThat(response.header("Expires")).isEqualTo("0");
+        assertThat(response.header("Access-Control-Allow-Credentials")).isEqualTo("true");
+        assertThat(response.header("Access-Control-Allow-Origin")).isEqualTo("http://www.example.com");
 
-        // facebook bid response for ad unit 5
-        wireMockRule.stubFor(post(urlPathEqualTo("/facebook-exchange"))
-                .withRequestBody(equalToJson(jsonFrom("auction/test-facebook-bid-request-1.json")))
-                .willReturn(aResponse().withBody(jsonFrom("auction/test-facebook-bid-response-1.json"))));
+        final String expectedAuctionResponse = auctionResponseFrom(jsonFrom("auction/facebook/test-auction-facebook-response.json"),
+                response, AUCTION_RESPONSE_TIME_PLACEHOLDER, Collections.singletonMap(FACEBOOK, "http://localhost:" + WIREMOCK_PORT + "/facebook-exchange"));
+        assertThat(response.asString()).isEqualTo(expectedAuctionResponse);
+    }
 
+    @Test
+    public void auctionShouldRespondWithBidsFromPulsepoint() throws IOException {
+        // given
         // pulsepoint bid response for ad unit 6
         wireMockRule.stubFor(post(urlPathEqualTo("/pulsepoint-exchange"))
-                .withRequestBody(equalToJson(jsonFrom("auction/test-pulsepoint-bid-request-1.json")))
-                .willReturn(aResponse().withBody(jsonFrom("auction/test-pulsepoint-bid-response-1.json"))));
+                .withRequestBody(equalToJson(jsonFrom("auction/pulsepoint/test-pulsepoint-bid-request-1.json")))
+                .willReturn(aResponse().withBody(jsonFrom("auction/pulsepoint/test-pulsepoint-bid-response-1.json"))));
+
+        // pre-bid cache
+        wireMockRule.stubFor(post(urlPathEqualTo("/cache"))//
+                .withRequestBody(equalToJson(jsonFrom("auction/pulsepoint/test-cache-pulsepoint-request.json")))
+                .willReturn(aResponse().withBody(jsonFrom("auction/pulsepoint/test-cache-pulsepoint-response.json"))));
+
+        // when
+        final Response response = given(spec)
+                .header("Referer", "http://www.example.com")
+                .header("X-Forwarded-For", "192.168.244.1")
+                .header("User-Agent", "userAgent")
+                .header("Origin", "http://www.example.com")
+                .cookie("uids",
```

Comment:
There is no need for this call to span 2 lines, please check throughout all tests